src/commands.ts

```diff
@@ -0,0 +1,168 @@
+
+    let isValid = await validateProjectSDK([project], commandObserver);
+    if (!isValid) {
+        return;
+    }
+    await dotnetBuild(dotNetSdk, project, commandObserver, cancelToken);
+}
+
+async function buildAllProjects(dotNetSdk: DotNetInfo, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    try {
+        let projects = await vscode.workspace.findFiles('{**/*.pgproj}');
+        commandObserver.clear();
+
+        let isValid = await validateProjectSDK(projects.map(p => p.fsPath), commandObserver);
+        if (!isValid) {
+            return;
+        }
+
+        for (let project of projects) {
+            if (cancelToken.isCancellationRequested) {
+                return;
+            }
+            await dotnetBuild(dotNetSdk, project.fsPath, commandObserver, cancelToken);
+        }
+    } catch (err) {
+        vscode.window.showErrorMessage(err);
+    }
+}
+
+async function validateProjectSDK(projects: string[], commandObserver: CommandObserver): Promise<boolean> {
+    var packageInfo = Utils.getPackageInfo();
+    let unsupportedProjects = await Helper.checkProjectVersion(packageInfo.minSupportedPostgreSQLProjectSDK, packageInfo.maxSupportedPostgreSQLProjectSDK, projects, commandObserver);
+    if (unsupportedProjects && unsupportedProjects.length > 0) {
+        unsupportedProjects.map(p => commandObserver.logToOutputChannel(strings.format(Constants.buildFailedUnsupportedSdkMessage, p)));
+        return Promise.resolve(false);
+    }
+    return Promise.resolve(true);
+}
+
+async function dotnetBuild(dotNetSdk: DotNetInfo, project: string, commandObserver: CommandObserver, cancelToken: vscode.CancellationToken): Promise<void> {
+    let args = ['build', project];
+    commandObserver.logToOutputChannel(strings.format(Constants.buildStartedMessage, project));
+    await runDotNetCommand(dotNetSdk, args, commandObserver, cancelToken);
+    commandObserver.logToOutputChannel(strings.format(Constants.buildCompletedMessage, project));
+}
+
+async function addNewPostgreSQLProject(args: vscode.Uri, dotNetSdk: DotNetInfo, projectTemplateNugedId: string, commandObserver: CommandObserver) {
+    let folder = args.fsPath;
+    var defaultProjectName = path.basename(folder);
+
+    var projectName = await vscode.window.showInputBox(
+        {
+            prompt: 'Project Name',
+            value: defaultProjectName,
+            validateInput: (value: string) => {
+                if (!value || value.trim().length === 0) {
+                    return Constants.projectNameEmptyErrorMessage;
+                }
+                if (/[\/?:&*"<>|#%;\\]/g.test(value))
+                {
+                    return Constants.projectNameSpecialCharsErrorMessage;
+                }
+                if (value === '.' || value === '..')
+                {
+                    return Constants.projectNameInvalidErrorMessage;
+                }
+                if (value.includes('..'))
```

Comment:
Hm.. is it true? I think ```foo..bar``` is still a valid folder/file name.

Reply:
I was trying to follow what Visual Studio allows and it does not allow you to create a project with '..' in name. I am trying to find why though